### PR TITLE
handle clicks on schema and examples

### PR DESCRIPTION
**Pull request by smacker**
Based on #50

Fix: #22 
Fix: #43

Examples aren't real. We need a list of examples from the product team.

### CODE REVIEW

File: frontend/src/components/SampleQueries.less [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.sample-queries {
+    .title {
+        padding: 3px 0;
+        margin-bottom: 10px;
+        background: #ccc;
+
```
Comment (dpordomingo) [inline frontend/src/components/SampleQueries.less]: extra space

File: frontend/src/components/Schema.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import PropTypes from 'prop-types';
+import { Glyphicon } from 'react-bootstrap';
+import './Schema.less';
+
+class SchemaTable extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { expanded: false };
+
```
Comment (dpordomingo) [inline frontend/src/components/Schema.js]: spaces?

File: frontend/src/components/Sidebar.less [hunk 3]
```diff
@@ -0,0 +1,34 @@
+.sidebar {
+    display: flex;
+    flex-direction: column;
+
+    min-height: 100%;
+
+    .header {
+        flex: 0 0;
+
```
Comment (dpordomingo) [inline frontend/src/components/Sidebar.less]: spaces?

File: frontend/src/components/SampleQueries.less (hunk 1 above)
Comment (smacker) [reply frontend/src/components/SampleQueries.less]: yes

File: frontend/src/components/Schema.js (hunk 2 above)
Comment (smacker) [reply frontend/src/components/Schema.js]: yes

File: frontend/src/components/Sidebar.less (hunk 3 above)
Comment (smacker) [reply frontend/src/components/Sidebar.less]: yes

File: frontend/src/components/Schema.js (hunk 2 above)
Comment (dpordomingo) [reply frontend/src/components/Schema.js]: are we applying any special rule/logic in behind?
I saw other quite condense files, so I wonder about the difference in between...

Comment (smacker) [reply frontend/src/components/Schema.js]: I put an empty line between logic blocks:
```
super() - first block

this.state = {} - second

lots of bindings - third
```

**Comment by smacker** [comment]
1. Hm. Current default SQL query doesn't work for me (gitbase schema is different, there is no repository_id field). Looks like the one you provided wouldn't also. I'm talking about this one:
```
SELECT COUNT(*) as num_commits, month, repo_id, committer_email FROM (
  SELECT MONTH(committer_when) as month, r.repository_id as repo_id, committer_email
  FROM repositories r
  INNER JOIN refs ON refs.repository_id = r.repository_id AND refs.ref_name = 'HEAD'
  INNER JOIN commits c ON YEAR(committer_when) = 2018 AND history_idx(refs.commit_hash, c.commit_hash) >= 0
) as t
GROUP BY committer_email, month, repo_id
```
And even if it will work with the next release of gitbase, I'm not sure that it's going to work in the release after next.

This PR implement frontend part of the sidebar interaction. I don't think invention of examples and catching up with gitbase which is in WIP state is in the scope of this PR.

2. 👍 (though it's weird that we don't have any handlers for them)

**Comment by carlosms** [comment]
Yes, the examples I suggested work only on master, but in my opinion that's better than the ones that don't work on either on master or the latest release. If you don't want to commit to any concrete examples in this PR, what about removing the current ones and just use `/* To be done */` for all of them?

**Pull request merged**